src/bun.js/webcore/response.zig

```diff
@@ -708,9 +720,100 @@ pub const Fetch = struct {
             this.javascript_vm.allocator.destroy(this);
         }
 
-        pub fn onDone(this: *FetchTasklet) void {
+        pub fn onBodyReceived(this: *FetchTasklet) void {
+            if (this.aborted.load(.Acquire)) return;
+
+            const success = this.result.isSuccess();
+            const globalThis = this.global_this;
+            defer {
+                if (!success or !this.result.has_more) {
+                    var vm = globalThis.bunVM();
+                    this.poll_ref.unref(vm);
+                    this.clearData();
+                    this.deinit();
+                }
+            }
+
+            if (!success) {
+                const err = this.onReject();
+                globalThis.throwValue(err);
+                return;
+            }
+
+            if (this.response) |response| {
```

Comment:
What is keeping this response alive? How can we be sure that the garbage collector doesn't free the Response object in the middle of streaming?

Reply:
fixed https://github.com/oven-sh/bun/pull/4127/commits/4676cfe8c877ca99e2fb4f2598afb51b51ceaadc